### PR TITLE
rudimentary cmake added

### PR DESCRIPTION
**Pull request by apmccartney**
No changes to the source code but added CMake integration to the repository. Assuming you have CMake 3.8 or greater installed, you can see a demonstration using the following:

```CMake
git clone https://github.com/apmccartney/frozen.git
cd frozen
mkdir build
cd build
cmake -D CMAKE_BUILD_TYPE=Release ..
make
make test
make install
```

As written, other cmake aware packages will be able to detect and consume an installation of the frozen headers as follows:

```CMake
find_package(frozen)
target_link_libraries(my_cool_library PUBLIC frozen::frozen)
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.8)
+get_directory_property(is_subproject PARENT_DIRECTORY)
+
+project(frozen VERSION 1.0.0 LANGUAGES CXX)
+include(CTest)
+include(CMakePackageConfigHelpers)
+include(CMakeDependentOption)
+
+option(frozen.cpp17
+  "Use the C++ 17 standard or greater for projects linking to frozen")
```
Comment (serge-sans-paille) [inline CMakeLists.txt]: Should be C++14

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+CMAKE_DEPENDENT_OPTION(frozen.tests
+  "Build the ndatk tests and integrate with ctest"
```
Comment (serge-sans-paille) [inline CMakeLists.txt]: ndatk :-)

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  "${BUILD_TESTING}" "${is_subproject}" OFF)
+
+CMAKE_DEPENDENT_OPTION(frozen.benchmark
+  "Build the ndatk tests and integrate with ctest"
```
Comment (serge-sans-paille) [inline CMakeLists.txt]: same

File: benchmarks/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+target_link_libraries(frozen.benchmark PUBLIC
+  frozen::frozen
+  benchmark::benchmark)
+
+target_sources(frozen.benchmark PRIVATE
+   ${CMAKE_CURRENT_LIST_DIR}/bench_int_set.cpp
+   ${CMAKE_CURRENT_LIST_DIR}/bench_main.cpp
+   ${CMAKE_CURRENT_LIST_DIR}/bench_str_set.cpp
+   $<$<COMPILE_FEATURES:cxx_std_17>:
+     ${CMAKE_CURRENT_LIST_DIR}/bench_str_search.cpp>)
```
Comment (serge-sans-paille) [inline benchmarks/CMakeLists.txt]: some of the benchmark files are generated by the makefile, you don't seem to have captured that aspect here.

**Comment by serge-sans-paille** [comment]
That's cool! You're much more proeficient in cmake than I am, I just added a few notes/remarks, once that's fixed I'll be happy to merge that PR!

File: CMakeLists.txt (hunk 1 above)
Comment (apmccartney) [reply CMakeLists.txt]: When this option is OFF, the C++ standard defaults to 14. I'll modify the doc string.

File: CMakeLists.txt (hunk 2 above)
Comment (apmccartney) [reply CMakeLists.txt]: ☺️

**Comment by apmccartney** [comment]
I beieve this is ready for a second-pass review =)

**Comment by serge-sans-paille** [comment]
LGTM, can you just update the README to reflect the change?

**Pull request merged**